eng/common/scripts/TypeSpec-Project-Process.ps1

```diff
@@ -0,0 +1,115 @@
+    $additionalDirs = $tspConfigYaml["dependencies"]["additionalDirectories"];
+  }
+
+  $packageDir = Get-PackageDir
+
+  $repoRootPath =  (Join-Path $PSScriptRoot .. ..)
+  $repoRootPath = Resolve-Path $repoRootPath
+  $repoRootPath = $repoRootPath -replace "\\", "/"
+
+  # Create service-dir if not exist
+  $serviceDir = Join-Path $repoRootPath "sdk" $serviceDir
+  if (!(Test-Path -Path $serviceDir)) {
+    New-Item -Path $serviceDir -ItemType Directory
+    Write-Host "created service folder $serviceDir"
+  }
+
+  # Create package-dir if not exist
+  $packageDir = Join-Path $serviceDir $packageDir
+  if (!(Test-Path -Path $packageDir)) {
+    New-Item -Path $packageDir -ItemType Directory
+    Write-Host "created package folder $packageDir"
+  }
+
+  # Load tsp-location.yaml if exist
+  $tspLocationYamlPath = Join-Path $packageDir "tsp-location.yaml"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    $tspLocationYaml = Get-Content -Path $tspLocationYaml -Raw | ConvertFrom-Yaml
+  }
+
+  # Update tsp-location.yaml
+  $tspLocationYaml["commit"] = $CommitHash
+  $repo = ($RepoUrl -split "/")[3..4] -join "/"
+  $tspLocationYaml["repo"] = $repo
+  $tspLocationYaml["directory"] = $TypeSpecProjectDirectory
+  $tspLocationYaml["additionalDirectories"] = $additionalDirs
+  $tspLocationYaml |ConvertTo-Yaml | Out-File $tspLocationYamlPath
+}
+
+function Get-PackageDir() {
+  if (Test-Path "Function:$GetEmitterNameFn") {
+    $emitterName = &$GetEmitterNameFn
+  }
+  $packageDir = ""
+  if ($tspConfigYaml["$emitterName"] -and $tspConfigYaml["$emitterName"]["package-dir"]) {
+    $packageDir = $tspConfigYaml["$emitterName"]["package-dir"];
+  }
+  else {
+    Write-Host "Missing package-dir in $emitterName options of tspconfig.yaml."
+    exit
+  }
+  return $packageDir
+}
+
+$tspConfigContent = ""
+# example url of tspconfig.yaml: https://github.com/Azure/azure-rest-api-specs-pr/blob/724ccc4d7ef7655c0b4d5c5ac4a5513f19bbef35/specification/containerservice/Fleet.Management/tspconfig.yaml
+if ($TypeSpecProjectDirectory -match '^https://github.com/.*/tspconfig.yaml$') {
+  try {
+    Invoke-WebRequest $TypeSpecProjectDirectory -OutFile $tspConfigContent -MaximumRetryCount 3
+  }
+  catch {
+    Write-Host "Failed to download '$TypeSpecProjectDirectory'"
+    Write-Error $_.Exception.Message
+    return
+  }
+  $RepoUrl = $TypeSpecProjectDirectory
+  $regex = "(?<=specification/).*?(?=/tspconfig.yaml)"
+  $TypeSpecProjectDirectory = [regex]::Match($TypeSpecProjectDirectory, $regex).Value
+  # TODO get the commit hash from the url
+} else {
+  $TspConfigPath = Join-Path $TypeSpecProjectDirectory "tspconfig.yaml"
+  if (!(Test-Path $TspConfigPath)) {
+    Write-Host "Failed to find tspconfig.yaml in '$TypeSpecProjectDirectory'"
+    return
+  }
+  $tspConfigContent = Get-Content $TspConfigPath -Raw
+}
+$tspConfigYaml = ConvertFrom-Yaml $tspConfigContent
+# call CreateUpdate-TspLocation function
+CreateUpdate-TspLocation
```

Comment:
You should pass parameters to this function instead of relying on global variables.